schema.graphql

```diff
@@ -1,18 +1,124 @@
-type BackerStakingHistory @entity {
+type BackerStakingHistory @entity(immutable: false) {
   id: Bytes!
-  backerTotalAllocation_: BigInt!
-  accumulatedTime_: BigInt!
-  lastBlockNumber_: BigInt!
-  lastBlockTimestamp_: BigInt!
-  gauges_: [GaugeStakingHistory!]! @derivedFrom(field: "backer_")
+  backerTotalAllocation: BigInt!
+  accumulatedTime: BigInt!
+  lastBlockNumber: BigInt!
+  lastBlockTimestamp: BigInt!
+  gauges: [GaugeStakingHistory!]! @derivedFrom(field: "backer")
 }
 
-type GaugeStakingHistory @entity {
+type GaugeStakingHistory @entity(immutable: false) {
   id: Bytes!
-  gauge_: Bytes! # address
-  backer_: BackerStakingHistory!
-  allocation_: BigInt!
-  accumulatedAllocationsTime_: BigInt!
-  lastBlockNumber_: BigInt!
-  lastBlockTimestamp_: BigInt!
+  gauge: Bytes! # address
+  backer: BackerStakingHistory!
+  allocation: BigInt!
+  accumulatedAllocationsTime: BigInt!
+  lastBlockNumber: BigInt!
+  lastBlockTimestamp: BigInt!
 }
+
+type Builder @entity(immutable: false) {
+  id: Bytes!
+  gauge: Bytes!
+  estimatedRewardsPct: BigDecimal!
+  backerRewardPercentage: BackerRewardPercentage! @derivedFrom(field: "builder")
+  rewardReceiver: Bytes!
+  state: BuilderState! @derivedFrom(field: "builder")
+  isHalted: Boolean!
+  lastCycleRewards: BigInt!
+  totalAllocation: BigInt!
+  rewardShares: BigInt!
+  proposals: [Proposal!]! @derivedFrom(field: "builder")
+  rewardsClaimed: [BuilderRewardsClaimed!]! @derivedFrom(field: "builder")
+}
+
+type BuilderState @entity(immutable: false) {
+  id: Bytes!
+  builder: Builder!
+  initialized: Boolean!
+  kycApproved: Boolean!
+  communityApproved: Boolean!
+  kycPaused: Boolean!
+  selfPaused: Boolean!
+  pausedReason: Bytes!
+}
+
+type BackerRewardPercentage @entity(immutable: false) { 
+  id: Bytes!
+  builder: Builder!
+  previous: BigInt!
+  next: BigInt!
+  cooldownEndTime: BigInt!
+}
+
+type GaugeToBuilder @entity(immutable: true) { 
+  id: Bytes!
+  builder: Bytes!
+}
+
+type BuilderRewardsClaimed @entity(immutable: false) {
+  id: Bytes!
+  token: Bytes!
+  amount: BigInt!
+  builder: Builder! 
+}
+
+
+type Proposal @entity(immutable: false) {
+  id: Bytes!
+  description: String!
+  link: String!
+  status: String!
+  proposalType: String!
+  builder: Builder!
+}
+
+type Backer @entity(immutable: false) {
+  id: Bytes!
+  isBlacklisted: Boolean!
+  totalAllocation: BigInt!
+  builders: [BackerToBuilder!] @derivedFrom(field: "backer")
+  rewardsClaimed: [BackersRewardsClaimed!] @derivedFrom(field: "backer")
+}
+
+type BackersRewardsClaimed @entity(immutable: false) {
+  id: Bytes!
+  token: Bytes!
+  amount: BigInt!
+  backer: Backer! 
```

Comment:
could we rename this to just `allocation`? Since this is basically same as `gauge.allocationOf(backer)`.

Reply:
@antomor should I rename it, it still called totalAllocation.

Reply:
For now, let's keep `totalAllocation`